tools/visual-pipeline-and-platform-evaluation-tool/Makefile

```
@@ -0,0 +1,110 @@
+# SPDX-FileCopyrightText: (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+SHELL	:= bash -eu -o pipefail
+
+# default goal to show help
+.DEFAULT_GOAL := help
+.PHONY: all lint build mdlint ruff style-fix mypy run build-dev run-dev test \
+	shell shell-vippet shell-models shell-collector shell-videogenerator stop clean help
+
+# Project Variables
+PROJECT_NAME := vippet
+PROJECT_DIR := $(shell pwd)
+
+# Docker Compose Variables
+COMPOSE_FILE := compose.yml
+DEV_COMPOSE_FILE := compose.dev.yml
+
+# Python venv Target
+VENV_DIR := .venv_vippet
+
+# Placeholder to prevent warnings about missing variables.
+# This value is overridden by setup_env.sh in relevant targets.
+export RENDER_GROUP_ID := 1000
+
+$(VENV_DIR): requirements.txt ## Create Python venv
+	python3 -m venv $@ ;\
+  set +u; . ./$@/bin/activate; set -u ;\
+  python -m pip install --upgrade pip ;\
+  python -m pip install -r requirements.txt
+
+all: lint build run test ## Run lint, build, run and test
+
+lint: $(VENV_DIR) mdlint ruff mypy ## Run all linters
+
+MD_FILES := $(shell find . -type f \( -name '*.md' \) -print )
+mdlint: ## Lint all markdown files
+	markdownlint --version
+	markdownlint $(MD_FILES)
+
+ruff: $(VENV_DIR) ## Run ruff linter
+	set +u; . ./$</bin/activate; set -u ;\
+	ruff --version ;\
+	ruff check
+	ruff format --check
+
+style-fix: $(VENV_DIR) ## Fix style issues with ruff
+	ruff check --fix
+	ruff format
+
+mypy: $(VENV_DIR) ## Run mypy type checker
+	mypy -V
+	mypy .
+
+check-device-type:
+	@if [ -z "$(DEVICE_TYPE)" ]; then \
+		echo "Error: DEVICE_TYPE is not set. Use 'make <target> DEVICE_TYPE=<GPU/CPU/NPU>'"; \
+		exit 1; \
+	fi; \
+
+build: check-device-type $(VENV_DIR) ## Build all images
+	source setup_env.sh -d $(DEVICE_TYPE) && \
+	docker compose build
+
+build-dev: check-device-type $(VENV_DIR) ## Build all dev images
+	source setup_env.sh -d $(DEVICE_TYPE) && \
+	docker compose -f $(COMPOSE_FILE) -f $(DEV_COMPOSE_FILE) build
+
+run: check-device-type ## Run the docker compose services
+	source setup_env.sh -d $(DEVICE_TYPE) && \
+	docker compose up -d
+
+run-dev: check-device-type $(VENV_DIR) ## Run the docker compose services for development
+	source setup_env.sh -d $(DEVICE_TYPE) && \
+	docker compose -f $(COMPOSE_FILE) -f $(DEV_COMPOSE_FILE) up -d
+
+test: ## Run tests
+	source setup_env.sh -d GPU && \
+	docker compose run --rm --no-deps --volume $(PROJECT_DIR):/home/dlstreamer/vippet vippet-gpu python -m unittest discover -v -s ./tests -p '*_test.py'
```

This shouldn't be hardcoded to GPU
```suggestion
	source setup_env.sh -d $(DEVICE_TYPE) && \
	docker compose run --rm --no-deps --volume $(PROJECT_DIR):/home/dlstreamer/vippet vippet-$(DEVICE_TYPE) python -m unittest discover -v -s ./tests -p '*_test.py'
```